system/Validation/StrictRules/FormatRules.php

```diff
@@ -0,0 +1,424 @@
+     *
+     * @param mixed|null $str
+     */
+    public function alpha_numeric($str = null): bool
+    {
+        if (is_int($str)) {
+            $str = (string) $str;
+        }
+
+        if (! is_string($str)) {
+            return false;
+        }
+
+        return $this->nonStrictFormatRules->alpha_numeric($str);
+    }
+
+    /**
+     * Alphanumeric w/ spaces
+     *
+     * @param mixed|null $str
+     */
+    public function alpha_numeric_space($str = null): bool
+    {
+        if (is_int($str)) {
+            $str = (string) $str;
+        }
+
+        if (! is_string($str)) {
+            return false;
+        }
+
+        return $this->nonStrictFormatRules->alpha_numeric_space($str);
+    }
+
+    /**
+     * Any type of string
+     *
+     * @param string|null $str
+     */
+    public function string($str = null): bool
+    {
+        return $this->nonStrictFormatRules->string($str);
+    }
+
+    /**
+     * Decimal number
+     *
+     * @param mixed|null $str
+     */
+    public function decimal($str = null): bool
+    {
+        if (is_int($str) || is_float($str)) {
+            $str = (string) $str;
+        }
+
+        if (! is_string($str)) {
+            return false;
+        }
+
+        return $this->nonStrictFormatRules->decimal($str);
+    }
+
+    /**
+     * String of hexidecimal characters
+     *
+     * @param mixed|null $str
+     */
+    public function hex($str = null): bool
+    {
+        if (is_int($str)) {
+            $str = (string) $str;
+        }
+
+        if (! is_string($str)) {
+            return false;
+        }
+
+        return $this->nonStrictFormatRules->hex($str);
+    }
+
+    /**
+     * Integer
+     *
+     * @param mixed|null $str
+     */
+    public function integer($str = null): bool
+    {
+        if (is_int($str)) {
+            $str = (string) $str;
+        }
+
+        if (! is_string($str)) {
+            return false;
+        }
+
+        return $this->nonStrictFormatRules->integer($str);
+    }
+
+    /**
+     * Is a Natural number  (0,1,2,3, etc.)
+     *
+     * @param mixed|null $str
+     */
+    public function is_natural($str = null): bool
+    {
+        if (is_int($str)) {
+            $str = (string) $str;
+        }
+
+        if (! is_string($str)) {
+            return false;
+        }
+
+        return $this->nonStrictFormatRules->is_natural($str);
+    }
+
+    /**
+     * Is a Natural number, but not a zero  (1,2,3, etc.)
+     *
+     * @param mixed|null $str
+     */
+    public function is_natural_no_zero($str = null): bool
+    {
+        if (is_int($str)) {
+            $str = (string) $str;
+        }
+
+        if (! is_string($str)) {
+            return false;
+        }
+
+        return $this->nonStrictFormatRules->is_natural_no_zero($str);
+    }
+
+    /**
+     * Numeric
+     *
+     * @param mixed|null $str
+     */
+    public function numeric($str = null): bool
+    {
+        if (is_int($str) || is_float($str)) {
```

Comment:
I have passed by many instances of this. How about using `is_numeric`?

Reply:
I don't agree with it. It checks the value type. `is_numeric()` is not for it.

And the return value of `is_numeric()` is not predictable. I don't want to use it.
https://3v4l.org/dbDev